src/scripts/core/models/background.ts

```diff
@@ -1,92 +1,175 @@
-        this._ship.position = {
-            x: this._viewWidth * 0.9 ,
-            y: this._viewHeight
-        }
-        this._ship.pivot = {
-            x: this._ship.width * 0.5, 
-            y: this._ship.height * 0.25
-        }
-        this._ship.angle = -20;
-        this._ship.alpha = 0;
-        gsap.to(this._ship, {
-            duration: 5,
-            repeat: -1,
-            yoyo: true,
-
-            y: this._ship.y + this._ship.height * 0.1,
-            angle: 20,
-        });
+        this._npcShipClose = this.addShip(0.4, 9);
+        this._npcShipClose.x = this._viewWidth * 0.3;
+        this._playerShip = this.addShip(0.6, 5);
+        this._playerShip.y = this._viewHeight - this._playerShip.height * 0.4
+        this._npcShipFar = this.addShip(0.3, 3);
+        
 
-        this._midWaveContainer.addChild(this._ship, this._midWave);
+        this._midWaveContainer.addChild(
+            this._npcShipFar,
+            this._playerShip,
+            this._npcShipClose,
+            this._midWave
+        );
 
         const background = new Sprite(Assets.get('background'));
         background.width = this._viewWidth;
         background.height = this._viewHeight;
         
-        this.addWavesToTimeline(0.2, 0.2, 0.2, 2);
+        this.addWavesToTimeline(0.15, 0.1, 0, 2);
         this.addChild(
-            background,
+            this._stormCloudsContainer,
             this._farWave,
             this._midWaveContainer,
-            this._closeWave
+            this._closeWave,
+            ground
         );
+
+        this.stopThunder();
+    }
+
+    public playThunder(): void {
+        this._thunderFlash.alpha = 1;
+        this._lightning.alpha = 1;
+        this._thunderFlash.state.setAnimation(0, "introduction", false);
+        this._lightning.state.setAnimation(0, "introduction", false);
+    }
+
+    public loopThunder(): void {
+        this._thunderFlash.alpha = 1;
+        this._lightning.alpha = 1;
+        this._thunderFlash.state.setAnimation(0, "introduction", true);
+        this._lightning.state.setAnimation(0, "introduction", true);
+    }
+
+    public stopThunder(): void {
+        this._thunderFlash.alpha = 0;
+        this._lightning.alpha = 0;
+        this._thunderFlash.state.setEmptyAnimation(0, 0);
+        this._lightning.state.setEmptyAnimation(0, 0);
+    }
+
+
+    private addShip(scale: number, duration: number): Sprite {
+        const ship = new Sprite(Assets.get('ship'));
+
+        ship.scale.set(scale);
+        ship.position = {
+            x: this._viewWidth * 0.8 ,
+            y: this._viewHeight - ship.height * 0.75
+        }
+        ship.pivot = {
+            x: ship.width * 0.75, 
+            y: ship.height 
+        }
+        ship.angle = -20;
+        ship.alpha = 0;
+        gsap.to(ship, {
+            duration: duration,
+            repeat: -1,
+            yoyo: true,
+            angle: 20,
+        });
+
+        return ship
```

Comment:
This function name implies you are adding the ship to the stage, but you're starting an animation before it has been added to the stage. I would argue this function should call addChild for the ship before you run the animation.

Heck it feels like the ships should be their own class which extends sprite, where the animations can be handled.